### PR TITLE
mlir attr types for infrt place (2nd PR), test=develop

### PR DESCRIPTION
**Pull request by Shixiaowei02**
### PR types

Others
### PR changes

Others
### Describe

supports the enum attrs, test=develop reference: https://github.com/PaddlePaddle/Paddle/pull/40087

### CODE REVIEW

**Review by Superjomn — DISMISSED** [review]
LGTM

**Review by winter-wang — DISMISSED** [review]
LGTM

File: paddle/infrt/dialect/phi/data_type.cc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/infrt/dialect/phi/data_type.h"
+
+namespace infrt {
+
+phi::Backend cvtTarget2Phi(TargetType target) {
```
Comment (Superjomn) [inline paddle/infrt/dialect/phi/data_type.cc:19]: Follow the google C style pls.

Comment (Shixiaowei02) [reply paddle/infrt/dialect/phi/data_type.cc:19]: I only moved the position of the function, I will fix it in the next PR

**Review by winter-wang — APPROVED** [review]
LGTM

**Pull request merged**